scripts/test262-runner.js

```diff
@@ -1088,6 +1129,82 @@ function runTest(
   }
 }
 
+function executeTestUsingSerializer(
+  test: TestFileInfo,
+  testFileContents: string,
+  data: BannerData,
+  // eslint-disable-next-line flowtype/no-weak-types
+  harnesses: Object,
+  strict: boolean,
+  timeout: number
+) {
+  let sources = [];
+
+  // Add the test262 intrinsics.
+  sources.push({
+    filePath: "test262.js",
+    fileContents: `\
+var $ = {
+  evalScript: () => {}, // noop for now
+  global,
+};
+var $262 = $;
+var print = () => {}; // noop for now
+  `,
+  });
+
+  // Add the harness files.
+  for (let name of ["sta.js", "assert.js"].concat(data.includes || [])) {
+    let harness = harnesses[name];
+    sources.push({ filePath: name, fileContents: harness });
+  }
+
+  // Add the test file.
+  sources.push({ filePath: test.location, fileContents: (strict ? '"use strict";' + EOL : "") + testFileContents });
+
+  let result;
+  try {
+    result = prepackSources(sources, {
+      serialize: true,
+      timeout: timeout * 1000,
+      errorHandler: diag => {
+        if (diag.severity === "Information") return "Recover";
+        if (diag.severity !== "Warning") return "Fail";
+        return "Recover";
+      },
+      onParse: () => {
+        // TODO(calebmer): Turn all literals into abstract values
+      },
+    });
+  } catch (error) {
+    if (error.message === "Timed out") return new TestResult(false, strict, error);
+    if (error.message.includes("Syntax error")) return null;
+    return new TestResult(false, strict, new Error(`Prepack error:\n${error.stack}`));
+  }
+
+  const context = vm.createContext({
+    // NOTE: Workaround since Prepack serializes code that expects a global `TypedArray` class which does not exist per
```

Comment:
Maybe it exists in JSC? Anyway, sounds like a bug that deserves an issue number and a TODO...

Reply:
#2292 👍